sdk/eventhub/event-hubs/src/log.ts

```diff
@@ -20,3 +20,62 @@ export function logErrorStackTrace(error: unknown): void {
     logger.verbose(error.stack);
   }
 }
+
+/**
+ * Creates a logger that includes the connectionId, sender or receiver name, and
+ * client type.
+ * @internal
+ */
+export function createLogPrefix(
+  connectionId?: string,
+  type?: "Sender" | "Receiver" | "Management",
+  name?: string
+): string {
+  let prefix = "";
+  if (connectionId) {
+    prefix = `[${connectionId}]`;
+  }
+  if (type) {
+    prefix += ` ${type}`;
+  }
+  if (name) {
+    prefix += ` [${name}]`;
+  }
+  return prefix;
+}
+
+type LogFunction = Debugger["log"];
+
+/**
+ * @internal
+ */
+export interface SimpleLogger {
```

Comment:
as an alternative, perhaps something like

```ts
export type SimpleLogger = {
  [Property in keyof AzureLogger]: Pick<Debugger, "log">
}
```

Seems a little simpler?

Reply:
Great catch! addressed in https://github.com/Azure/azure-sdk-for-js/pull/25214/commits/9e1faf1551c63799b5ad35689ebb081699067fa1.